src/routes/jira/jira-get-connections-backfillStatus.ts

```diff
@@ -0,0 +1,150 @@
+import { NextFunction, Request, Response } from "express";
+import { groupBy } from "lodash";
+import { RepoSyncState } from "~/src/models/reposyncstate";
+import {
+	TaskStatus,
+	Subscription,
+	SyncStatus
+} from "~/src/models/subscription";
+import { mapSyncStatus, ConnectionSyncStatus } from "./jira-get";
+
+type SubscriptionBackfillState = {
+	totalRepos?: number;
+	isSyncComplete: boolean;
+	syncedRepos?: number;
+	backfillSince?: string | null;
+	syncStatus: ConnectionSyncStatus;
+};
+type BackFillType = {
+	[key: string]: SubscriptionBackfillState;
+};
+
+export const JiraGetConnectionsBackfillStatus = async (
+	req: Request,
+	res: Response,
+	next: NextFunction
+): Promise<void> => {
+	try {
+		const { jiraHost: localJiraHost } = res.locals;
+		const subscriptionIds: number[] = String(req.query?.subscriptionIds)
+			.split(",")
+			.map((id) => Number(id))
+			.filter(Boolean);
+
+		if (subscriptionIds?.length <= 0) {
+			req.log.warn("Missing Subscription IDs");
+			res.status(400).send("Missing Subscription IDs");
+			return;
+		}
+		const subscriptions: Subscription[] = await Subscription.findAll({
+			where: {
+				id: subscriptionIds
+			}
+		});
+
+		if (subscriptions?.length <= 0) {
+			req.log.error("Missing Subscription");
+			res.status(400).send("Missing Subscription");
+			return;
+		}
+
+		const jiraHosts: string[] = subscriptions.map(
+			(subscription) => subscription?.jiraHost
+		);
+
+		const jiraHostsMatched = jiraHosts.every(
+			(jiraHost) => jiraHost === localJiraHost
+		);
+
+		const subscriptionsById = groupBy(subscriptions, "id");
+		if (!jiraHostsMatched) {
+			req.log.error("Missmached Jira Host");
+			res.status(403).send("Missmached Jira Host");
+			return;
+		}
+
+		const repoSyncStates = await RepoSyncState.findAll({
+			where: {
+				subscriptionId: subscriptionIds
+			}
+		});
+
+		const repos = groupBy(repoSyncStates, "subscriptionId");
+
+		const backfillStatus = getBackfillStatus(repos, subscriptionsById);
+
+		const isBackfillComplete = getBackfillCompletionStatus(backfillStatus);
+		res.status(200).send({
+			data: {
+				subscriptions: backfillStatus,
+				isBackfillComplete,
+				subscriptionIds
+			}
+		});
+	} catch (error) {
+		return next(new Error(`Failed to render connected repos: ${error}`));
+	}
+};
+
+const getBackfillCompletionStatus = (backfillStatus: BackFillType): boolean => {
+	let isBackfillComplete = true;
```

Comment:
Don't need to assign a default value. Can directly do 
` return Object.values(backfillStatus).every(
		(backFill: SubscriptionBackfillState): boolean => backFill?.isSyncComplete
	);`

Reply:
corrected